include/particle.tcc

```diff
@@ -389,9 +389,12 @@ bool mpm::Particle<Tdim, Tnphases>::map_mass_momentum_to_nodes(unsigned phase) {
 
 // Compute strain of the particle
 template <unsigned Tdim, unsigned Tnphases>
-void mpm::Particle<Tdim, Tnphases>::compute_strain(unsigned phase, double dt) {
+void mpm::Particle<Tdim, Tnphases>::compute_strain(unsigned phase, double dt,
+                                                   bool bbar) {
   // Strain rate
-  const auto strain_rate = cell_->compute_strain_rate(bmatrix_, phase);
+  auto strain_rate = cell_->compute_strain_rate(bmatrix_, phase);
+  if (bbar) strain_rate = cell_->compute_strain_rate_bbar(bmatrix_, phase);
```

Comment:
I cannot declare just within if-statements. This is my current solution - any feedback is appreciated

Reply:
@kks32, what do you think about this?

Reply:
Declare the type and initalize based on which one is being used